### PR TITLE
Start 'Street Explorer' test browser for viewing tests

### LINKED ISSUE #13: Create a slippy map interface with layers served by the tile server
We want to set up a tile server #12 which can render our data, so;

It would be nice to have a slippy map with those layers to cycle through at a press.

### LINKED ISSUE #22: Create a slippy map "Tests Browser"
This repo contains an independent test suite that tests`RoadNetwork` and abstreet's `RawMap`. Each test case is a real world example with a lot of geographic data.

We should build a slippy map that shows the various formats of data for a test case, layering the .osm with the .geojson and linking the .dot. A simple menu would let you choose a test case to view, and highlight tests that need approving.

It could then be extended to run tests on demand, downloading .osm on the fly.

### PR DESCRIPTION
**Pull request by BudgieInWA**
Works towards #22, closes #13 by introducing a `trunk serve` slippy map with controls, that (will hopefully soon) shows the test data from `tests/`.

TODO
- [x] Allow trunk to read the tests dir to list test cases and load test case files
- [x] Add GeoJSON from a test file to the map (as proof of concept)
- ~~Generate the test list at build time, link to the `.json` files~~

CC @droogmic @dabreegster

### CODE REVIEW

**Comment by dabreegster** [comment]
Heads-up, you'll need to rebase against 3fac825a38fd9ae3ed0fd0384a792cc71e089018, which simplified build deps a bit. It should be straightforward.

It's awesome to see this come along! I'm curious how you find yew; does it work well as a way of writing web apps? If you were using pure JS or TypeScript, what would your typical tech stack of choice be?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Working on osm2streets
+
+Come hang out at https://github.com/a-b-street/osm2streets to ask questions and improve these docs.
+
+## Pre-requisites
+- rust beta
```
Comment (droogmic) [inline CONTRIBUTING.md:6]: rust nightly?

Comment (droogmic) [reply CONTRIBUTING.md:6]: I suggest just encoding this using https://github.com/a-b-street/osm2lanes/blob/main/rust-toolchain.toml

File: streets-web/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+license = "Apache 2.0"
+edition = "2021"
+
+[dependencies]
+console_log = { version = "0.2", optional = true, features = ["color"] }
+getrandom = { version = "0.2", features = ["js"] }
+gloo-utils = "0.1"
+leaflet = { git = "https://github.com/droogmic/leaflet-rs", branch = "polyline_get_bounds" }
+log = "0.4"
+#osm2lanes = { path = "../osm2lanes", features = ["overpass"] }
```
Comment (droogmic) [inline streets-web/Cargo.toml:16]: I'll start working at dropping the dependencies you need out of the bottom of osm2lanes :)

**Review by droogmic — APPROVED** [review]
Apologies for the delay, I missed your ping for my attention :)

I suggest already preparing the continuous deployment to github.io, even if still unfinished.

**Comment by droogmic** [comment]
> Heads-up, you'll need to rebase against [3fac825](https://github.com/a-b-street/osm2streets/commit/3fac825a38fd9ae3ed0fd0384a792cc71e089018), which simplified build deps a bit. It should be straightforward.
> 
> It's awesome to see this come along! I'm curious how you find yew; does it work well as a way of writing web apps? If you were using pure JS or TypeScript, what would your typical tech stack of choice be?

+1 on this, as I am not a web / full stack dev. I approached yew as a systems software engineer.

**Comment by BudgieInWA** [comment]
> It's awesome to see this come along! I'm curious how you find yew; does it work well as a way of writing web apps? If you were using pure JS or TypeScript, what would your typical tech stack of choice be?

Through circumstance more than anything, I have worked a lot with React in [Meteor](https://www.meteor.com/) projects. I like the declarative style of React, and being consistent with the flow of data.

Looking at the osm2lanes source, yew looks pretty neat. The "event -> message -> state transition -> view calls" workflow seems sensible to me, and being in rust land will bring advantages, right? Like being able to call the impl directly.

The first hurdle will be for me to learn how to interface with existing JS. It looks like I could write the bindgen translation thingo to call into `L.OSM.DataLayer`, or `openFile` or whatever function I want to write? A big part of this tends to be sticking together different pieces, so having some JS land code might be appropriate alongside the yew app, but idk.

File: CONTRIBUTING.md (hunk 1 above)
Comment (BudgieInWA) [reply CONTRIBUTING.md:6]: rust beta seems to be working for me, if I understand my setup right, so I'd like to stay there if possible. I'll add the `rust-toolchain.toml`.

**Comment by BudgieInWA** [comment]
> Apologies for the delay, I missed your ping for my attention :)

:)

> I suggest already preparing the continuous deployment to github.io, even if still unfinished.

Yes, great idea! #26

**Comment by BudgieInWA** [comment]
I will leave these for the next PR:

- Generate the test list at build time to link to each test.
- Adding a writeup to the README showing it off and sharing future plans.

**Pull request merged**